### ISSUE TITLE
Is the file gold_labels.tsv mannually created?

### ISSUE THREAD

**Issue by aindrilabasak** [HazyResearch/snorkel#941]
Hello,

I am working with Snorkel and trying to apply it on a large tabular dataset. Right now I have created the gold label records in the gold_labels.tsv file manually, and the code seems to work. My question is, is it the right way to create the gold_labels? Or is there an automated way, with which I can extract the gold labels from my dataset? As my dataset has more than a million record, and it will be hard to create the gold labels manually.

**Comment by bjcmit** [comment]
Manually creating the gold labels is the right way. If there was an automated way to create highly accurate labels (i.e. gold labels), why would you not use that to label all your data? Finally, would you please share how you applied Snorkel to a large tabular dataset?

**Comment by aindrilabasak** [comment]
I think my question was confusing. I know that the gold labels are manually created labels provided by the domain experts. I wanted to ask, if the file gold_labels.tsv in the tutorial 'Extracting Spouse Relations from the News' is manually created? As this file has a specific structure such as 'DocumentID::span:start:end   Label'. So I wanted to know, are these start and end positions calculated manually?

Also, as of now, I have applied Snorkel on an extremely small portion of my tabular dataset.  I followed the idea from https://github.com/HazyResearch/snorkel/issues/446 issue, and created my own candidate hierarchy. Although, I am at the preliminary phase with it. However, there are many challeges that need to be addressed at this point. One of the examples can be, memory issues (discussed in: https://github.com/HazyResearch/snorkel/issues/936).

**Comment by stephenbach** [comment]
Just one other thought. It sounds like you might be asking about just preprocessing the data. Certainly we don't type gold_labels.tsv by hand, but writing code to produce it is specific to each data set and beyond the scope of Snorkel. Consider checking out the Pandas library for creating tables and writing them to TSV.

**Comment by aindrilabasak** [comment]
@stephenbach thank you for your answer, I figured it out some time ago. I didn't know how to close the issue. :)
@bjcmit Thank you. To answer your question on how did I apply snorkel on a tabular data, I just converted my pandas dataframe to a .tsv file with all the features separated with the delimiter specified in the TSVDocPreprocessor. About the dataset size, I used the guidelines mentioned in https://github.com/HazyResearch/snorkel/issues/936.
